includes/discovery/sensors/state/fortigate.inc.php

```diff
@@ -36,31 +36,51 @@
     // Fetch the cluster members
     $haStatsEntries = snmpwalk_cache_multi_oid($device, $fgHaStatsIndex_txt, [], 'FORTINET-FORTIGATE-MIB');
 
+    // As of July 2024:
+    // FortiNet considers a single-node cluster a valid setup. Therefor, if a cluster node is removed or
+    // defective (lost power, defective hardware) no error message can be seen. The FortiGate silently
+    // removes the cluster member from the configuration after a few minutes and the cluster status becomes OK
+
+    // Additionally, a FortiGate cluster is always in-sync with the primary.
+    // So fgHaStatsSyncStatus.1 (primary node) will *always* return 1. FortiNet does not consider this
+    // a bug (reference: Fortinet ticket 9671105)
+
+    // So if we have a 2-node cluster and either one of the nodes goes offline, no alert or warning
+    // can be seen in the UI. The sync status will remain 1 (In-Sync) for node/index 1 and index 2
+    // will cease to exist. Again, Fortinet considers this expected behavior.
+
+    // Use the 'Cluster State' sensor to create an alert when this value drops to 1.
+    // -- Rudy Broersma
+
+    // Per node sync status
+    $stateName = 'fgHaStatsSyncStatus';
+
+    $states = [
+        ['value' => 0, 'generic' => 2, 'graph' => 0, 'descr' => 'Out Of Sync'],
+        ['value' => 1, 'generic' => 0, 'graph' => 1, 'descr' => 'In Sync'],
+    ];
+
+    create_state_index($stateName, $states);
+
+    $fgHaStatsSyncStatus_num = '.1.3.6.1.4.1.12356.101.13.2.1.1.12';
+    $fgHaStatsSyncStatus_txt = 'fgHaStatsSyncStatus';
+
+    // Per-node loop
     foreach ($haStatsEntries as $index => $entry) {
         // Get name of cluster member
         $fgHaStatsHostname_txt = 'fgHaStatsHostname.' . $index;
         $cluster_member_name = snmp_get($device, $fgHaStatsHostname_txt, '-Ovq', 'FORTINET-FORTIGATE-MIB');
 
-        // Setup a sensor for the cluster sync state
-        $fgHaStatsSyncStatus_num = '.1.3.6.1.4.1.12356.101.13.2.1.1.12';
-        $fgHaStatsSyncStatus_txt = 'fgHaStatsSyncStatus';
-        $stateName = 'haSyncStatus ' . $index;
         $descr = 'HA sync status ' . $cluster_member_name;
 
-        $states = [
-            ['value' => 0, 'generic' => 2, 'graph' => 0, 'descr' => 'Out Of Sync'],
-            ['value' => 1, 'generic' => 0, 'graph' => 1, 'descr' => 'In Sync'],
-            ['value' => 2, 'generic' => 1, 'graph' => 0, 'descr' => 'No Peer'],
-        ];
-
-        create_state_index($stateName, $states);
+        // Setup a sensor for the cluster sync state
 
         discover_sensor(
             $valid['sensor'],
             'state',
             $device,
             $fgHaStatsSyncStatus_num . '.' . $index,
-            $fgHaStatsSyncStatus_txt . '.' . $index,
```

Comment:
Any reason you removed $fgHaStatsSyncStatus_txt ?

Reply:
Yes, if I put that back the state indexes no longer work and the sensors get the status 'OK' again (instead of In Sync or Out of Sync). So it breaks the state index translations.

Additionally if I look at the PHP example at the bottom of https://docs.librenms.org/Developing/Sensor-State-Support/#advanced-example it shows:

discover_sensor($valid['sensor'], 'state', $device, $num_oid.$index, $index, $state_name,

In that example parameter 5 is also just the index, not the text representation of the OID plus index.
Using just the index for parameter 5 seems to be the way to go according to the documentation?